apps/homescreen/test/unit/datastore_test.js

```diff
@@ -0,0 +1,370 @@
+                          },
+                          continue: () => {
+                          }
+                        }
+                      }
+                    });
+                  }
+                };
+              }
+            };
+          },
+          set oncomplete(cb) {
+            cb();
+          },
+          onerror: () => {
+          }
+        };
+      };
+    };
+
+    setup(() => {
+      bookmarks.db = {};
+    });
+
+    suite('Datastore#set()', () => {
+      var data = {
+        id: 'abc/',
+        misc: 'def'
+      };
+
+      setup(() => {
+        attachDBTransaction(bookmarks.storeName);
+      });
+
+      test('should store data', done => {
+        bookmarks.set(data)
+          .then(() => {
+            assert.equal(Object.keys(objectStores).length, 1);
+            assert.equal(objectStores[bookmarks.storeName].id, data.id);
+            assert.equal(objectStores[bookmarks.storeName].data, data);
+            done();
+          });
+      });
+
+      test('should emit an event', done => {
+        document.addEventListener(bookmarks.name + '-set', evt => {
+          assert.equal(evt.detail.id, data.id);
+          done();
+        });
+        bookmarks.set(data);
+      });
+    });
+
+    suite('Datastore#remove()', () => {
+      var id = 'abc/';
+
+      setup(() => {
+        attachDBTransaction(bookmarks.storeName);
+      });
+
+      test('should remove data from object store', done => {
+        bookmarks.remove(id)
+          .then(() => {
+            assert.deepEqual(deletedValues, [id]);
+            done();
+          });
+      });
+
+      test('should emit an event', done => {
+        document.addEventListener(bookmarks.name + '-removed', evt => {
+          assert.equal(evt.detail.id, id);
+          done();
+        });
+        bookmarks.remove(id);
+      });
+    });
+
+    suite('Datastore#clear()', () => {
+      test('should remove all data from object store', done => {
+        attachDBTransaction(bookmarks.storeName);
+        bookmarks.clear()
+          .then(() => {
+            done();
+          });
+      });
+
+      test('should emit an event', done => {
+        attachDBTransaction(bookmarks.storeName);
+        document.addEventListener(bookmarks.name + '-cleared', () => {
+          done();
+        });
+        bookmarks.clear();
+      });
+    });
+
+    suite('Datastore#get()', () => {
+      test('should retrieve data from object store', done => {
+        attachDBTransaction(bookmarks.storeName);
+        bookmarks.get('abc/')
+          .then(results => {
+            assert.deepEqual(results, { id: 'abc/', data: {} });
+            done();
+          });
+      });
+    });
+
+    suite('Datastore#getAll()', () => {
+      test('should retrieve data from object store', done => {
+        attachDBTransaction(bookmarks.storeName);
+        bookmarks.getAll()
+          .then(results => {
+            assert.equal(results.length, 1);
```

Comment:
It feels like for this test we should add more than 1 entry in the store to make sure that it returns all of them.